cmd/ib-sriov-cni/main.go

```diff
@@ -82,6 +108,14 @@ func cmdAdd(args *skel.CmdArgs) error {
 	}
 
 	sm := sriov.NewSriovManager()
+
+	// Lock CNI operation to serialize the operation
+	lock, err := lockCNIExecution()
```

Comment:
it should be noted, that while further optimizations where possible both here and in CmdDel (unlock eralier than on first error or end of execution)
it would have introduced (imo) un-needed code complexity